lib/contentful/resource/custom_resource.rb

```diff
@@ -0,0 +1,19 @@
+module Contentful
+  module Resource
+    module CustomResource
+      def initialize(*)
+        super
+
+        unless client.nil?
+          if client.dynamic_entry_cache.key?(content_type['sys']['id'].to_sym)
```

Comment:
Should we warn/fail if `client == nil` or the content type isn't cached, yet? Might be better than silently not having any fields.